bundles/http/Citrix/Citrix-Virtual-Apps-and-Desktops-service/7687284a-09a1-4494-b33c-f6bfa0932656.js

```diff
@@ -0,0 +1,671 @@
+    fields,
+    body,
+) {
+    let path = `${siteId}/${entity}/$search?fields=${fields}&limit=${limit}`
+
+    if (continuationToken) {
+        path = `${path}&continuationToken=${continuationToken}`
+    }
+    const response = await client.fetch(path, {
+        method: 'POST',
+        body: body,
+        headers: {
+            'Content-Type': 'application/json',
+            'Citrix-CustomerId': integrationParameters.customerId,
+        },
+    })
+
+    if (!response.ok) {
+        throw new Error(
+            `Endpoint ${entity} - Request failed(${response.status}: ${response.statusText})`,
+        )
+    }
+    console.log(entity + ' - response received, status: ' + response.status)
+
+    return response.json()
+}
+
+function flatJson(json, oName) {
+    if (json[oName] != undefined) {
+        Object.entries(json[oName]).map(([key, value]) => (json[oName + '_' + key] = value))
+        delete json[oName]
+    }
+    return json
+}
+
+integration.define({
+    synchronizations: [
+        {
+            name: 'sync',
+            fullSyncFunction: fullSync,
+            incrementalSyncFunction: incrementalSync,
+        },
+    ],
+    model: {
+        tables: [
+            {
+                name: 'machine',
+                columns: [
+                    {
+                        name: 'AgentVersion',
+                        type: 'STRING',
+                        length: 255,
+                    },
+                    {
+                        name: 'DeliveryGroup_Id',
+                        type: 'STRING',
+                        length: 255,
+                    },
+                    {
+                        name: 'DeliveryGroup_Name',
+                        type: 'STRING',
+                        length: 255,
+                    },
+                    {
+                        name: 'DnsName',
+                        type: 'STRING',
+                        length: 255,
+                    },
+                    {
+                        name: 'FaultState',
+                        type: 'STRING',
+                        length: 255,
+                    },
+                    {
+                        name: 'Id',
+                        type: 'STRING',
+                        length: 255,
+                        primaryKey: true,
+                    },
+                    {
+                        name: 'InMaintenanceMode',
+                        type: 'BOOLEAN',
+                    },
+                    {
+                        name: 'IPAddress',
+                        type: 'STRING',
+                        length: 255,
+                    },
+                    {
+                        name: 'LastConnectionFailure',
+                        type: 'STRING',
+                        length: 255,
+                    },
+                    {
+                        name: 'LastConnectionTime',
+                        type: 'STRING',
+                        length: 255,
```

Comment:
maybe `DATETIME `?
Same for the rest of "xxxTime" columns

Reply:
Had to add `new Date()`, but done.